### PR TITLE
Dealing with redirected updateContext requests due to polling

### PR DESCRIPTION
**Pull request by gtorodelvalle**
- 100% tests passed
- Assigned to @dmoranj

### CODE REVIEW

File: lib/context_adapter_communication.js [hunk 1]
```diff
@@ -183,6 +183,22 @@ function checkUpdateContextRequest(request, callback) {
...
 /**
+ * Returns true if the updateContext request corresponds to a polling request from the button
+ * @param {object} request The received request
+ * @return {boolean} true if the updateContext request corresponds to a polling request from the button,
+ *  false otherwise
+ */
+function isPollingUpdateContext(request) {
+  if (request && request.payload && request.payload.contextElements && request.payload.contextElements[0]) {
+    if (caHelper.getAttributeValue(
+        request.payload.contextElements[0].attributes, caConfig.BUTTON_ENTITY.CA_LAST_OPERATION) === 'P') {
```
Comment (AlvaroVega) [inline lib/context_adapter_communication.js:194]: Just the first one? I mean first element of contextElements

File: lib/context_adapter_communication.js [hunk 2]
```diff
@@ -518,6 +534,22 @@ function getThirdPartyPayload(serviceDescriptor, operationDescriptor) {
...
 
 /**
+ * Encodes the response sent to the Context Broker so no forbidden characters are used
+ *  The forbidden characters not allowed by the Context Broker are: < > " ' = ; ( )
+ * @param str The string to be encoded
+ * @return {string} The encoded string
+ */
+function encodeResponse(str) {
+  // encodeURIComponent() escapes all characters except the following: alphabetic, decimal digits, - _ . ! ~ * ' ( )
+  var encodedStr = encodeURIComponent(str);
```
Comment (dmoranj) [inline lib/context_adapter_communication.js]: I'm not sure it's a good idea to URL encode the response you are going to use to fill a JSON that no one expects to be URL Encoded... You will end up with an unreadable messy text full of control characters in a log or in a text box, as no one will decode it.

File: lib/context_adapter_communication.js (hunk 1 above)
Comment (gtorodelvalle) [inline lib/context_adapter_communication.js:194]: Yeah, for the time being only one request (info) will come in each updateContext, as far as I know? :)

File: lib/context_adapter_communication.js (hunk 2 above)
Comment (gtorodelvalle) [inline lib/context_adapter_communication.js]: Well, the problem are the forbidden characters not allowed by the Context Broker and the fact that the response is dynamic (not only the errors which may be sent to the Context Broker but also the result which may come from the Third Party service) :)
We could only encode these forbidden characters and not using `encodeURIComponent()` and everything would be much more readable (in the logs :) ).
What do you think?

Comment (dmoranj) [inline lib/context_adapter_communication.js]: I tend to think the better option is to remove the forbidden characters.

Comment (gtorodelvalle) [inline lib/context_adapter_communication.js]: Fixed in https://github.com/telefonicaid/context-adapter/commit/cb846a6008d6351915189eb17dab241bdd0e4108

**Pull request merged**